working-groups/query-standardization/dsl-survey/ppl/README.md

```diff
@@ -0,0 +1,256 @@
+
+In Response to the [DSL Survey requirements](https://github.com/cncf/tag-observability/tree/main/working-groups/query-standardization/dsl-survey) the next document addresses these concerns and emphasize the advantages of PPL language that should become the best option matching the new Observability query language:
+
+### **Short Description:**
+
+PPL (Pipe Processing Language) is a query language used in OpenSearch, allowing users to interact with their data in a more expressive and readable manner. It features a structure similar to Unix's pipe commands, where operations are chained together using a pipe symbol (|). This design enables users to perform complex queries in a more intuitive way, such as filtering, transforming, and aggregating data. By leveraging PPL, developers can enhance their ability to extract insights from large datasets within the OpenSearch ecosystem.
+
+The PPL (Pipe Processing Language) architecture within OpenSearch is built to facilitate querying and data manipulation in a way that resembles Unix's pipe commands.
+
+### Overview
+
+1. **Design Goals**:
+
+    * **URI Friendliness**: Enables intuitive querying through RESTful API.
+    * **Minimal Syntax**: Human-readable and concise.
+    * **Advanced Analytical Capabilities**: Supports aggregation, filtering, etc.
+    * **Easily extensible :** support simple and build in extension mechanics
+
+1. **Deficiencies in Other DSLs**:
+
+    * A need for a simpler, more consistent querying language led to the design of PPL.
+    * Emphasize the pipe based data transformation patterns common to realtime analytics and batch procrssing
+
+1. **Languages that Inspired the DSL**:
+
+    * SQL, and other structured query languages.
+    * Splunk Query Language
+
+1. **Observability Use Cases Designed For**:
+
+    * Analyzing logs, metrics, and tracing data.
+
+1. **Observability Use Cases Omitted**:
+
+    * Joins are currently not supported in a full extent - Work In Progress
+
+1. **Strengths and Weaknesses**:
+
+    * **Strengths**: Flexibility, ease of learning.
+    * **Weaknesses**: May lack some specific functionalities of other DSLs.
+
+1. **Intended for At-Rest and/or Streaming Data**:
+
+    * Both.
+
+1. **Source Specification**:
+
+    * Users can specify indices or aliases or generic data-sources that abstract the tables as a logical raw data .
+
+1. **Joining with Other Sources**:
+
+    * Joining multiple indices is supported; joining with external sources like CSV is not yet supported.
+
+1. **Differing Retentions in Time Series Sources**:
+
+    * Query federation from Prometheus is supported
+    * This would depend on the OpenSearch data management setup.
+
+1. **Coupling to the Data Store Model (1 to 10)**:
+
+    * 4, It's designed for OpenSearch but with some level of abstraction.
+
+1. **Flexibility Across Different Storage Formats or Contexts**:
+
+    * PPL is specific to OpenSearch and the underlying data structures it supports.
```

Comment:
If you don't mind, would ya'll be able to document the models in OS/ES under https://github.com/cncf/tag-observability/tree/main/working-groups/query-standardization/telemetry-models please?